packages/algorand-js/src/types.ts

```diff
@@ -11,6 +11,109 @@ export type AppArgs = Array<string | number>;
 export type StackElem = bigint | Uint8Array;
 export type TEALStack = IStack<bigint | Uint8Array>;
 
+// https://developer.algorand.org/docs/reference/teal/opcodes/#txn
+export enum TxnField {
```

Comment:
I think we already have transaction field in algosdk types. Let's use it.

Reply:
- This is an `enum` not an interface (are enums present in types/algosdk)?
- I couldn't find an interface also. Can you be more specific?

Reply:
how are we planning to use this enum in context ? 
`mkTransaction` will return `transaction` interface.
User cannot pass enum, mkTransaction will not pass enum
how `txn` field will return required value ?

you can search `Transaction`. there is a interface with this name in index.d.ts in types@algosdk.

Reply:
We are already using `Transaction` interface in the `createContext` function. 
`TxnField` enum is used to represent the transaction fields. (you can check txn opcode spec)
Use Case:
 - TEAL:
```
...
txn Fee
...
```
- PARSER:
```
push `new Txn(TxnField.Fee, this.interpreter)` to logic[]
```
- INTERPRETER 
```
reads this.interpreter.tx and pushes Fee to stack (as the txfield is Fee) 
```  
Let me know if that's clear

Reply:
what i am trying to ask here is:
- we get transaction interface from mkTransaction
- interpreter will push Fee to stack
- how are we connecting enum with interface because currently they are disconnected, (you will get blank value for each field)
does this makes sense ?

Reply:
Yes. We will have to create a function which will return specified value from transaction. This should be done because the TEAL opcode specification for transaction fields are different (in naming) than the `Transaction` interface of sdk. This would be done while adding the txn related opcodes. 
Something like:
```
function getTxnInfo(field: TxnField, transaction: Transaction) {
  switch (TxField) {
   case Fee:
        return transaction.fee
   ... 
   ...
 }
}
```
Should this be handled some other way?

Reply:
I was thinking of switching with strings here, this will result in less repetitive code (because we still have to switch with strings in parser)
but we have `TypeEnum`, i think handling with enum is better.

Reply:
yes, using enums is better than using strings.